src/test/java/com/limechain/network/protocol/grandpa/GrandpaEngineTest.java

```diff
@@ -0,0 +1,238 @@
+                38, 117, 16, -19, 19, -47, 73, -43, -103, -3, 115, 35, -50, 58, 61, 118, -102, -107, 32, 47, -79, 60,
+                85, 7, -72, 60, 8, 118, 4, 1, 0, 0 };
+        CommitMessage commitMessage = new CommitMessage();
+        commitMessage.setRoundNumber(BigInteger.valueOf(15283));
+        commitMessage.setSetId(BigInteger.valueOf(1573));
+        commitMessage.setVote(new Vote());
+        commitMessage.getVote().setBlockHash(
+                Hash256.from("0x477f65a7f7e1267510ed13d149d599fd7323ce3a3d769a95202fb13c5507b83c"));
+        commitMessage.getVote().setBlockNumber(BigInteger.valueOf(17069576));
+        commitMessage.setPrecommits(new Precommit[]{});
+
+        when(stream.isInitiator()).thenReturn(false);
+        when(CONNECTION_MANAGER.isGrandpaConnected(peerId)).thenReturn(true);
+
+        grandpaEngine.receiveRequest(message, stream);
+
+        verify(SYNCED_STATE).syncCommit(commitMessage, peerId);
+    }
+
+    @Test
+    void receiveNeighbourMessageOnResponderStreamWhenShouldSyncNeighbourMessage() {
+        byte[] message = new byte[] { 2, 1, -24, 60, 0, 0, 0, 0, 0, 0, 37, 6, 0, 0, 0, 0, 0, 0, -37, 118, 4, 1 };
+        NeighbourMessage neighbourMessage = new NeighbourMessage(1, BigInteger.valueOf(15592),
+                BigInteger.valueOf(1573), BigInteger.valueOf(17069787));
+
+        when(stream.isInitiator()).thenReturn(false);
+        when(CONNECTION_MANAGER.isGrandpaConnected(peerId)).thenReturn(true);
+
+        grandpaEngine.receiveRequest(message, stream);
+
+        verify(SYNCED_STATE).syncNeighbourMessage(neighbourMessage, peerId);
+    }
+    @Test
+    void receiveVoteMessageOnResponderStreamShouldLogAndIgnore() {
+        byte[] message = new byte[] { 0, 2, 3 };
+        when(stream.isInitiator()).thenReturn(false);
+        when(CONNECTION_MANAGER.isGrandpaConnected(peerId)).thenReturn(true);
+
+        grandpaEngine.receiveRequest(message, stream);
+
+        verify(LOGGER).log(eq(Level.INFO), contains("Vote message received"));
+    }
+
+    @Test
+    void receiveCatchUpRequestMessageOnResponderStreamShouldLogAndIgnore() {
+        byte[] message = new byte[] { 3, 2, 3 };
+        when(stream.isInitiator()).thenReturn(false);
+        when(CONNECTION_MANAGER.isGrandpaConnected(peerId)).thenReturn(true);
+
+        grandpaEngine.receiveRequest(message, stream);
+
+        verify(LOGGER).log(eq(Level.INFO), contains("Catch up request received"));
+    }
+
+    @Test
+    void receiveCatchUpResponseMessageOnResponderStreamShouldLogAndIgnore() {
+        byte[] message = new byte[] { 4, 2, 3 };
+        when(stream.isInitiator()).thenReturn(false);
+        when(CONNECTION_MANAGER.isGrandpaConnected(peerId)).thenReturn(true);
+
+        grandpaEngine.receiveRequest(message, stream);
+
+        verify(LOGGER).log(eq(Level.INFO), contains("Catch up response received"));
+    }
```

Comment:
These tests only test whether the warning is logged but not if the function returns.